### PR TITLE
added email validation for signUp users in Platform

### PR DESCRIPTION
**Pull request by haykeminyan**
Users previously could use fake non-email fields like 123, 321 e.t.c

Example curl:

```
curl --location 'https://account.bold.ru/' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0' \
--header 'Accept: */*' \
--header 'Accept-Language: en' \
--header 'Accept-Encoding: gzip, deflate, br' \
--header 'Referer: https://app.bold.ru/' \
--header 'Content-Type: application/json' \
--header 'Origin: https://app.bold.ru' \
--header 'Connection: keep-alive' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-site' \
--header 'Sec-GPC: 1' \
--header 'TE: trailers' \
--data '{"method":"createAccount","params":["1234","4343","test","test"]}'
```

now added from Svelte dev validation for email link [https://svelte.dev/repl/54d159b954d9412c8247807125d9fe1b?version=3.12.1]

### CODE REVIEW

**Comment by haiodo** [comment]
Thanks for a contribution, could you please add sign off and gpg signature. Thanks.

**Comment by haykeminyan** [comment]
> Thanks for a contribution, could you please add sign off and gpg signature. Thanks.

thanks! have used this [link](https://driesvints.medium.com/using-a-gpg-key-to-sign-off-git-commits-and-emails-9c2b44a914b5) for adding GPG and Sign-Off. Checked with
`git log --show-signature`
seems now is a good.

**Comment by haiodo** [comment]
Hi, you need to signoff and sign all commits, please use following git command to do so.

```git rebase --exec 'git commit --amend --no-edit -n -s -S' origin/main```

File: plugins/login-resources/src/utils.ts [hunk 1]
```diff
@@ -76,6 +76,11 @@ export async function doLogin (email: string, password: string): Promise<[Status
   }
 }
 
+function emailValidator (value: string): boolean {
```
Comment (haiodo) [inline plugins/login-resources/src/utils.ts:82]: We also need same logic inside server/account to prevent direct API call.

**Comment by haykeminyan** [comment]
> `git rebase --exec 'git commit --amend --no-edit -n -s -S' origin/main`

I have tried but seems failing.
❯ git rebase --exec 'git commit --amend --no-edit -n -s -S' origin/main
Executing: git commit --amend --no-edit -n -s -S
error: gpg failed to sign the data
fatal: failed to write commit object
warning: execution failed: git commit --amend --no-edit -n -s -S
You can fix the problem, and then run

  git rebase --continue

After executing command with --continue get:

git rebase --continue
Executing: git commit --amend --no-edit -n -s -S
interactive rebase in progress; onto 715193957
Last commands done (4 commands done):
   pick aad6fc42a added email validation # empty
   exec git commit --amend --no-edit -n -s -S
  (see more in file .git/rebase-merge/done)
No commands remaining.
You are currently editing a commit during a rebase.



No changes
You asked to amend the most recent commit, but doing so would make
it empty. You can repeat your command with --allow-empty, or you can
remove the commit entirely with "git reset HEAD^".
warning: execution failed: git commit --amend --no-edit -n -s -S
You can fix the problem, and then run

  git rebase --continue

File: plugins/login-resources/src/utils.ts (hunk 1 above)
Comment (haykeminyan) [reply plugins/login-resources/src/utils.ts:82]: + have noticed that this problem exists not only in SIgnUp method. In LogIn exists this mini-bug too :=)

curl log

`curl 'https://account.bold.ru/' -X POST -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0' -H 'Accept: */*' -H 'Accept-Language: en' -H 'Accept-Encoding: gzip, deflate, br' -H 'Referer: https://app.bold.ru/' -H 'Content-Type: application/json' -H 'Origin: https://app.bold.ru' -H 'Connection: keep-alive' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-site' -H 'Sec-GPC: 1' -H 'TE: trailers' --data-raw '{"method":"login","params":["123","123"]}'`

File: plugins/login-resources/src/utils.ts [hunk 2]
```diff
@@ -67,6 +67,9 @@ export async function doLogin (email: string, password: string): Promise<[Status
       },
       body: JSON.stringify(request)
     })
+    if (!emailValidator(email)) {
```
Comment (haykeminyan) [inline plugins/login-resources/src/utils.ts:70]: this logic contains the same as previous, I just observe that logIn function needs to be updated as signUp too.

**Comment by haykeminyan** [comment]
> > `git rebase --exec 'git commit --amend --no-edit -n -s -S' origin/main`
> 
> I have tried but seems failing. ❯ git rebase --exec 'git commit --amend --no-edit -n -s -S' origin/main Executing: git commit --amend --no-edit -n -s -S error: gpg failed to sign the data fatal: failed to write commit object warning: execution failed: git commit --amend --no-edit -n -s -S You can fix the problem, and then run
> 
> git rebase --continue
> 
> After executing command with --continue get:
> 
> git rebase --continue Executing: git commit --amend --no-edit -n -s -S interactive rebase in progress; onto [7151939](https://github.com/hcengineering/platform/commit/715193957b76be02ff0e5e0e4be82b2964aa35c7) Last commands done (4 commands done): pick aad6fc42a added email validation # empty exec git commit --amend --no-edit -n -s -S (see more in file .git/rebase-merge/done) No commands remaining. You are currently editing a commit during a rebase.
> 
> No changes You asked to amend the most recent commit, but doing so would make it empty. You can repeat your command with --allow-empty, or you can remove the commit entirely with "git reset HEAD^". warning: execution failed: git commit --amend --no-edit -n -s -S You can fix the problem, and then run
> 
> git rebase --continue

I have dropped previous commits and now contains only 1 signed commit. Is it good now?)

File: plugins/login-resources/src/utils.ts (hunk 1 above)
Comment (haykeminyan) [reply plugins/login-resources/src/utils.ts:82]: @haiodo I have added it for the server side [commit](https://github.com/hcengineering/platform/pull/4637/commits/7d9b5a83f8a6dab0e352b059721622d06e4a3c22) as the same for UI, but as I haven't deep knowledge of your product could you please check?)

**Pull request closed without merging**